### PR TITLE
refactored eks module to idf project_name, added unit-tests

### PR DESCRIPTION
**Pull request by srinivasreddych**
*Issue #, if available:*

refactored eks module to idf project_name, added unit-tests 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: data/eks_dockerimage-replication/versions/1.21.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+ami:
```
Comment (dgraeber) [inline data/eks_dockerimage-replication/versions/1.21.yaml:1]: Is there a need to support 1.21 going foward?

File: data/eks_dockerimage-replication/versions/1.22.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+ami:
+  version: 1.22.17-20230217
```
Comment (dgraeber) [inline data/eks_dockerimage-replication/versions/1.22.yaml:2]: Is there a need to support 1.22 going forward?

File: modules/compute/eks/docs/eks-isolated/eks-isolated-eksmanifest.yaml [hunk 3]
```diff
@@ -0,0 +1,117 @@
+name: eks
+path: modules/core/eks/
+dataFiles:
```
Comment (dgraeber) [inline modules/compute/eks/docs/eks-isolated/eks-isolated-eksmanifest.yaml:3]: The local reference breaks the gitpath functionality, doesn't it?  How would this module deploy from a gitpath definition?

File: data/eks_dockerimage-replication/versions/1.21.yaml (hunk 1 above)
Comment (srinivasreddych) [reply data/eks_dockerimage-replication/versions/1.21.yaml:1]: There is no support from eks for any versions under 1.23 ATM. So deleting them

File: data/eks_dockerimage-replication/versions/1.22.yaml (hunk 2 above)
Comment (srinivasreddych) [reply data/eks_dockerimage-replication/versions/1.22.yaml:2]: There is no support from eks for any versions under 1.23 ATM. So deleting them

**Review by dgraeber — APPROVED** [review]

File: modules/compute/eks/app.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    deployment_name=deployment_name,
+    module_name=module_name,
+    vpc_id=vpc_id,
+    controlplane_subnet_ids=controlplane_subnet_ids,
+    dataplane_subnet_ids=dataplane_subnet_ids,
+    eks_version=eks_version,
+    eks_compute_config=eks_compute_config,
+    eks_addons_config=eks_addons_config,
+    custom_subnet_ids=custom_subnet_ids,
+    codebuild_sg_id=codebuild_sg_id if os.getenv("SEEDFARMER_PARAMETER_CODEBUILD_SG_ID") else None,
```
[the remark this answers is not in the document]
Comment (srinivasreddych) [reply modules/compute/eks/app.py:68]: Where are the 2 places?

File: modules/compute/eks/app.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    replicated_ecr_images_metadata=replicated_ecr_images_metadata
```
Comment (dgraeber) [inline modules/compute/eks/app.py:69]: `replicated_ecr_images_metadata` is being set in 2 places...above and here

File: modules/compute/eks/app.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (dgraeber) [reply modules/compute/eks/app.py:68]: here and next line...

**Review by dgraeber — APPROVED** [review]

**Pull request merged**